### PR TITLE
Add null equality to EQ JPA expression visitor

### PR DESCRIPTION
**Pull request by tinawu0603**
**1. Issue Link:** 
https://github.com/intuit/graphql-filter-java/issues/23

**2. Brief explanation of a change:** 
Instead of assuming all values of `eq` to use `criteriaBuilder.equal`, if the value is `null`, we want to use `criteriaBuilder.isNull` instead to create the right specification.

**3. Will it break existing clients and code in production?**
No

### CODE REVIEW

**Comment by tinawu0603** [comment]
@jaiswalsk would you be able to review this PR?

**Comment by jaiswalsk** [comment]
Thank you @tinawu0603 for the pull request. I will review this and share my feedback soon! Thanks.

**Comment by tinawu0603** [comment]
@jaiswalsk Do you have an update for this PR? 🙏

File: pom.xml [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
   <groupId>com.intuit.graphql</groupId>
   <artifactId>graphql-filter-java</artifactId>
-  <version>1.0.0-SNAPSHOT</version>
```
Comment (jaiswalsk) [inline pom.xml:9]: Can you please revert this changes, as it will be taken care during the release process.

**Review by jaiswalsk — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (tinawu0603) [reply pom.xml]: Just reverted, thank you!

**Review by jaiswalsk — APPROVED** [review]
Looks good. Thank you!
